File: code/charlie/Pick6.py
import random

def pick6():
    temp = [random.randint(1, 100) for num in range(6)]
    temp.sort()
    return temp

def win_matches(win_nums, my_ticket, winnings):
    num_matches = 0
    for i in range(6):
        if win_nums[i] == my_ticket[i]:
            num_matches += 1
    
    if num_matches in winnings:
        return winnings[num_matches]

    return 0

def main():
    winnings = {6: 25000000, 5: 1000000, 4: 50000, 3: 100, 2: 7, 1: 4, 0: 0}
    win_nums = pick6()
    my_ticket = pick6()
    balance = 0
    earnings = 0
    expenses = 0

    for i in range(100000):
        earnings += win_matches(win_nums, my_ticket, winnings)
        balance -= 2
        expenses += 2
        win_nums = pick6()
        my_ticket = pick6()

    ROI = (earnings - expenses)/expenses
    balance += earnings
    print("Earnings = ", earnings)
    print(f'Your balance after 100,000 plays = {balance}')
    print('Your ROI for this round = ', ROI)
main()

File: code/charlie/JavaScript/lab1_p1/pick6.js
function compareNumbers(a, b){
    return a - b;
}
function pick6(){
    let temp = [];
    for(let i = 0; i < 6; ++i){
        let num = Math.random() * 100;

        if(num >= 1){
            temp[i] = Math.floor(num);
        } else{
            temp[i] = Math.ceil(num);
        }
    }
    // console.log(temp);
    temp.sort(compareNumbers);
    // console.log(temp);
    return temp;
}

function num_matches(my_nums, win_nums, winnings){
    let matches = 0;
    
    for(let i = 0; i < 6; ++i){
        if(my_nums[i] === win_nums[i]){
            ++matches;
        }
    }

    return winnings.payout[matches];    
}

let balance = 0;
let earnings = 0;
let expenses = 0;
let winnings ={
    payout: [0, 4, 7, 100, 50000, 1000000, 25000000]
    
};
let my_nums = pick6();
let win_nums = pick6();

for(let i = 0; i < 100000; ++i){
    earnings += num_matches(my_nums, win_nums, winnings);
    balance -= 2;
    expenses += 2;
    my_nums = pick6();
    win_nums = pick6();
}

let ROI = (earnings - expenses)/expenses;
balance += earnings;
alert(`Earnings = ${earnings}`);
alert(`Your balance after 100,000 plays = ${balance}`);
alert(`Your ROI for this round = ${ROI}`);

//alert(`Here are your random numbers for the lotto6: ${my_nums}`)